### ISSUE TITLE
sparse_tensor_to_dense() does not seem to support back propagation

### ISSUE THREAD

**Issue by dblueeye** [tensorflow/tensorflow#22543]
**Have I written custom code: Yes
**OS Platform and Distribution: Windows 10 (primary), Ubunto 16.04.3 LTS (secondary)
**TensorFlow installed from (source or binary): Binary
**TensorFlow version (use command below): b'v1.10.0-rc1-19-g656e7a2b34' 1.10.0
**Python version: Python 3.6.6 :: Anaconda, Inc.
CUDA/cuDNN version: N/A (not relevant)
GPU model and memory: N/A (not relevant)

### Describe the problem
The function sparse_tensor_to_dense() does not seem to support backpropagation. I've written a test script below, which yielded 

> Fetch argument None has invalid type <class 'NoneType'>

If you change the line
`grad_A00 = tf.gradients(A_dense_transformed[0,0], par)`
to the next (currently commented-out)
`grad_A00 = tf.gradients(A_dense_manual[0,0], par)`
the script can be executed.

If the sparse tensor formulation is necessary, the workaround is to use the commented lines 
```
identity = tf.Variable(eye(2), dtype = float32)
A_dense_transformed = A = tf.sparse_tensor_dense_matmul(tf.sparse_transpose(A_sparse), identity)
```
to replace original `A_dense_transformed = A = tf.sparse_tensor_to_dense(A_sparse)`. 
Such an exercise allows the program to execute the original `grad_A00 = tf.gradients(A_dense_transformed[0,0], par)`.

### Source code / logs
```

import tensorflow as tf
from numpy import *
# showing backprop fails if using sparse tensor

tf.reset_default_graph()

par = tf.Variable(1.0, dtype = float32)
A_sparse = tf.SparseTensor(indices=[[0,0],], values=tf.stack([par,]), dense_shape = (2,1))

A_dense_manual = tf.stack([[par],[0]])
A_dense_transformed = A = tf.sparse_tensor_to_dense(A_sparse)
#identity = tf.Variable(eye(2), dtype = float32)
#A_dense_transformed = A = tf.sparse_tensor_dense_matmul(tf.sparse_transpose(A_sparse), identity)

grad_A00 = tf.gradients(A_dense_transformed[0,0], par)
#grad_A00 = tf.gradients(A_dense_manual[0,0], par)

init = tf.global_variables_initializer()

with tf.Session() as sess:
    
    init.run()
    vA_manual, vA_transformed, vGrad = sess.run([A_dense_manual, A_dense_transformed, grad_A00])    
    
print(vA_manual)
print(vA_transformed)
print(vGrad)
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
Exact command to reproduce
Mobile device

**Comment by dblueeye** [comment]
> Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
> Bazel version
> Exact command to reproduce
> Mobile device

Thank you for your inquiry.
* I installed from binary, and not from source, so Bazel version does not matter. 
* The exact command to reproduce is just by simply running the attached source script (simple and short), `python script.py`
* The mobile device doesn't apply to my case: I did not test on mobile devices.

**Comment by wt-huang** [comment]
@dblueeye You are correct that sparse tensor works for back propagation when you use `tf.Variable`. We will add the improved functionality in the future.

**Comment by dblueeye** [comment]
Wait... what do you mean, @wt-huang? Are you claiming sparse_tensor_to_dense() works for back propagation, and you did not reproduce my error message?

**Comment by dblueeye** [comment]
@wt-huang Thanks for the clarification!
